File: MauiDashboard/Models/Ticket.cs
namespace TicketSystem.Maui.Models;

public class Ticket
{
    public int Id { get; set; }
    public string Titulo { get; set; } = string.Empty;
    public string Descripcion { get; set; } = string.Empty;
    public string Estado { get; set; } = "Abierto";
    public string Prioridad { get; set; } = "Media";
    public string ClienteNombre { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaActualizacion { get; set; }
}

File: FlutterApp/lib/models/ticket.dart
class Ticket {
  final int? id;
  final String titulo;
  final String descripcion;
  final String estado;
  final String prioridad;
  final String clienteNombre;
  final DateTime? fechaCreacion;

  Ticket({
    this.id,
    required this.titulo,
    required this.descripcion,
    required this.estado,
    required this.prioridad,
    required this.clienteNombre,
    this.fechaCreacion,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['id'],
      titulo: json['titulo'],
      descripcion: json['descripcion'],
      estado: json['estado'],
      prioridad: json['prioridad'],
      clienteNombre: json['clienteNombre'],
      fechaCreacion: json['fechaCreacion'] != null 
          ? DateTime.parse(json['fechaCreacion']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'titulo': titulo,
      'descripcion': descripcion,
      'estado': estado,
      'prioridad': prioridad,
      'clienteNombre': clienteNombre,
      'fechaCreacion': fechaCreacion?.toIso8601String(),
    };
  }
}
